src/test/scala/ch/epfl/bluebrain/nexus/tests/NewBaseSpec.scala

```diff
@@ -0,0 +1,214 @@
+              error.`@type`.getOption(json) shouldBe Some("NothingToBeUpdated")
+            case s => fail(s"We were not expecting $s when setting default acls for service account")
+          }
+      }
+    }
+  }
+
+  // Init the given realm
+  def initRealm(realm: String, identity: Identity): Assertion = {
+    cl.get[Json](s"/realms/$realm", identity) {
+      (json, response) =>
+        response.status match {
+          case StatusCodes.NotFound =>
+            logger.info(s"Realm $realm is absent, we create it")
+            val body =
+              jsonContentOf(
+                "/iam/realms/create.json",
+                Map(
+                  quote("{realm}") -> s"${config.realmSuffix(realm)}"
+                )
+              )
+            cl.put[Json](s"/realms/$realm", body, identity) {
+              (_, response) =>
+                response.status shouldEqual StatusCodes.Created
+            }
+            cl.get[Json](s"/realms/$realm", Identity.ServiceAccount) {
+              (_, response) =>
+                response.status shouldEqual StatusCodes.OK
+            }
+          case StatusCodes.Forbidden | StatusCodes.OK =>
+            logger.info(s"Realm $realm has already been created, we got status ${response.status}")
+            cl.get[Json](s"/realms/$realm", Identity.ServiceAccount) {
+              (_, response) =>
+                response.status shouldEqual StatusCodes.OK
+            }
+          case s => fail(s"$s wasn't expected here and we got this response: $json")
+        }
+    }
+  }
+
+  private[tests] implicit class HttpClientOps(val httpClient: UntypedHttpClient[Task]) {
+
+    def post[A](url: String, body: Json, identity: Identity)
+               (assertResponse: (A, HttpResponse) => Assertion)
+               (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(POST, url, Some(body), identity)(assertResponse)
+
+    def put[A](url: String, body: Json, identity: Identity)
+              (assertResponse: (A, HttpResponse) => Assertion)
+              (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(PUT, url, Some(body), identity)(assertResponse)
+
+    def patch[A](url: String, body: Json, identity: Identity)
+              (assertResponse: (A, HttpResponse) => Assertion)
+              (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(PATCH, url, Some(body), identity)(assertResponse)
+
+    def get[A](url: String, identity: Identity)
+              (assertResponse: (A, HttpResponse) => Assertion)
+              (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(GET, url, None, identity)(assertResponse)
+
+    def delete[A](url: String, identity: Identity)
+                 (assertResponse: (A, HttpResponse) => Assertion)
+                 (implicit um: FromEntityUnmarshaller[A]): Assertion =
+      request(DELETE, url, None, identity)(assertResponse)
+
+    def request[A](method: HttpMethod,
```

Comment:
using something like https://github.com/djspiewak/cats-effect-testing would allow us to avoid blocking for a result with automatic handling of timeouts... thoughts?